lib/kafo/hook_context.rb

```diff
@@ -65,6 +65,11 @@ def module_enabled?(module_name)
       !mod.nil? && mod.enabled?
     end
 
+    # Check if --disable-system-checks was passed by the user
+    def disable_system_checks?
```

Comment:
Currently nothing reads `skip_checks_i_know_better?` so that ends up being ignored in most cases.

Reply:
`skip_checks_i_know_better?` is read still at https://github.com/theforeman/kafo/pull/255/files#diff-657b82e4fb9c80f11badef608b14de10R137 which is the original behavior

my intention is to use `disable_system_checks?` helper in foreman-installer in place of the old `--disable-system-checks` hook (i.e. to disable other hooks)

as I have this currently written, the user would notice the following similarities/differences:

1. `--skip-checks-i-know-better` should behave exactly the same as before, skipping only the checks but not any hooks, with the caveat that it prints a deprecation warning

2. `--disable-system-checks` would not only disable the same hooks it already does, but now will disable the checks as well. since we are planning to merge the checks into hooks, this feels like the correct behavior for the new combined option

I could have this helper return true for `--skip-checks-i-know-better` as well but i'd rather just have it continue the old behavior and print the deprecation warning which gives the suggestion to use the new combined option. is this agreeable to you @ekohl ?

Reply:
Kafo is a library and every now and then we do get an issue reported so it's used outside of foreman-installer as well. That means it needs to present a consistent UI/API.

I'd suggest there is just a single way to use checks as a consumer (mostly hooks, but also API). That's why I'm suggesting clear methods as hooks.

Next is the compatibility with older versions.

From an API perspective, right now there's `kafo.skip_checks_i_know_better?` that's callable in hooks (see `katello/hooks/pre_commit/05-puppet_certs_exist.rb` in current foreman-installer`). I think you can also check it via `app_value(:skip_checks_i_know_better)` but I don't see an example for that.

From the UI perspective there's `--skip-checks-i-know-better`.

You can choose the compatibility you want to keep. Since this likely ends up as a breaking change and a major version, IMHO it's fine to keep none. If needed, the consumers of kafo can implement compatibility via hooks. You can even provide an upgrade example in docs. The benefit is that you don't carry it in kafo.

If you want to keep UI compatibility, you can catch `--skip-checks-i-know-better` and trigger skipping all checks.

API compatibility is also possible, but I think it'd end up ugly, complex and of minor value.

Reply:
I originally suggested to @wbclark to keep `--skip-checks-i-know-better` compatibility for an amount of time in order to keep that for the community used to it. If we think that this is a minor thing and it's easier to drop it and simply communicate to users to change to the new syntax then I'd be happy to drop that and simplify things for developers.

Reply:
So the conclusion I gathered from this discussion is for @wbclark to drop `skip-checks-i-know-better` and go with `disable-system-checks` or even just `disable-checks` since we will major version bump Kafo with this change anyway.

Reply:
Yes, but if we're going to rename, we should really provide a proper implementation. Just renaming from `--skip-checks-i-know-better` to `--skip-checks` provides no value IMHO. If desired, foreman-installer can already drop `--disable-system-checks` and standardize on the `--skip-checks-i-know-better`.

I really don't want to rename things twice and there's no rush to IMHO.